### ISSUE TITLE
Query: [Nav Prop Translation] SelectMany generates wrong SQL

### ISSUE THREAD

**Issue by srollinet** [aspnet/EntityFramework#3889]
When I execute this query with the RC1 version
```c#
var test1 = context.Sports
	.Where(s => s.Id == 1)
	.SelectMany(s => s.Teams)
	.SelectMany(s => s.Players)
	.Select(p => p.Name)
	.ToList();
```
The generated SQL looks like
```sql
SELECT [s].[Id], [s].[Name], [s].[TeamId], [s].[Id], [s0].[Id], [p].[Name]
FROM [Player] AS [s]
CROSS JOIN [Team] AS [s0]
CROSS JOIN [Player] AS [p]
WHERE [s].[Id] = @__s1_Id_0
```

1) the tables and columns are wrong (s should be Sport)
2) the join conditions are missing -> cartesian product
3) less important, but too many columns are selected. Only [p].[Name] is really usefull

and with this query
```c#
var test2 = context.Teams
	.Where(t => t.SportId == 1)
	.SelectMany(t => t.Players)
	.Select(p => p.Name)
	.ToList();
```
The generated SQL looks like
```sql
SELECT [t].[Id], [t].[Name], [t].[TeamId], [t].[Id], [p].[Name]
FROM [Player] AS [t]
CROSS JOIN [Player] AS [p]
WHERE [t].[SportId] = @__s1_Id_0
```
Same problems, but here an exception is thrown because [t].[SportId] doesn't exist

**Comment by vuorinem** [comment]
+1 (just ran into this issues with very similar query)

**Comment by mejje** [comment]
Is there a known workaround for this bug?

**Comment by srollinet** [comment]
You can join manually rather than using the navigation properties.

```c#
var test1b = (from s in context.Sports
			  join t in context.Teams on s.Id equals t.SportId
			  join p in context.Players on t.Id equals p.TeamId
			  where s.Id == 1
			  select p.Name).ToList();
```
has the expected result

**Comment by adrien-constant** [comment]
Any ETA on this?

The workaround works, but this prevents code reuse and that's a pain to rewrite the code with manual joins when you want to call SelectMany.

**Comment by rowanmiller** [comment]
@adrien-constant this is on our list of issues to fix for our next pre-release (it's being fixed as part of a larger area of work on navigation property support in query)

**Comment by maumar** [comment]
fixed in 7d27bf6527391cd46dbb09210f7e40f7998e549e
